uui-core/src/data/processing/views/__tests__/Tree.patch.test.ts

```diff
@@ -0,0 +1,204 @@
+        const newItem = { id: 400, name: 'some new item' };
+        const patchedTree = tree.patch([newItem], 'isDeleted');
+        expect(patchedTree['byId'].has(400)).toBeTruthy();
+        expect(patchedTree === tree).toBeFalsy();
+        expect(patchedTree['byId'].get(400)).toEqual(newItem);
+    });
+
+    it('should add item into exact place', () => {
+        const newItem = { id: 335, name: 'some new item', parentId: 300 };
+        const patchedTree = tree.patch(
+            [newItem],
+            'isDeleted',
+            ({ id }, { id: existingId }) => (id > existingId) ? 1 : -1,
+        );
+        expect(patchedTree['byId'].has(335)).toBeTruthy();
+        expect(patchedTree === tree).toBeFalsy();
+        expect(patchedTree['byId'].get(335)).toEqual(newItem);
+        expect(patchedTree['byParentId'].get(300)).toEqual([
+            310, 320, 330, 335, 340, 350,
+        ]);
+    });
+
+    it('should add, update and delete items during patch', () => {
+        const deletedItem = { id: 350, parentId: 300, isDeleted: true };
+        const updatedItem = { id: 310, parentId: 300, name: 'new name' };
+        const newItem = { id: 335, name: 'some new item', parentId: 300 };
+        const patchedTree = tree.patch(
+            [deletedItem, updatedItem, newItem],
+            'isDeleted',
+            ({ id }, { id: existingId }) => (id > existingId) ? 1 : -1,
+        );
+        expect(patchedTree['byId'].has(deletedItem.id)).toBeFalsy();
+        expect(patchedTree['byId'].has(updatedItem.id)).toBeTruthy();
+        expect(patchedTree['byId'].has(newItem.id)).toBeTruthy();
+
+        expect(patchedTree['byId'].get(updatedItem.id)).toEqual(updatedItem);
+        expect(patchedTree['byId'].get(newItem.id)).toEqual(newItem);
+        expect(patchedTree === tree).toBeFalsy();
+
+        expect(patchedTree['byParentId'].get(newItem.parentId)).toEqual([
+            310, 320, 330, 335, 340,
+        ]);
+    });
+
+    it('should not update tree if items were not updated', () => {
+        const updatedItem = testData[1];
+        const patchedTree = tree.patch([updatedItem]);
+        expect(patchedTree['byId'].has(updatedItem.id)).toBeTruthy();
+        expect(patchedTree['byId'].get(updatedItem.id)).toEqual(updatedItem);
+        expect(patchedTree === tree).toBeTruthy();
+
+        expect(patchedTree['byParentId'].get(updatedItem.parentId)).toEqual([110, 120]);
+
+        const updatedItem2 = testData[2];
+        const patchedTree2 = tree.patch(
+            [updatedItem2],
+            'isDeleted',
+            ({ id }, { id: existingId }) => (id > existingId) ? 1 : -1,
+        );
+        expect(patchedTree2['byId'].has(updatedItem2.id)).toBeTruthy();
+        expect(patchedTree2['byId'].get(updatedItem2.id)).toEqual(updatedItem2);
+        expect(patchedTree2 === tree).toBeTruthy();
+
+        expect(patchedTree2['byParentId'].get(updatedItem.parentId)).toEqual([110, 120]);
+    });
```

Comment:
Let's add test(s) to check that:
- we can insert 2 new items to the top of the list, with default comparator. Items should be inserted in the order they present in array: patch = [1,2] , tree looks like [2,1,...existing]
- we can insert 2 new items at the bottom of the list. Patch = [1,2], comparator: () => 1, tree: [...existing, 1, 2,]